### DISCUSSION TITLE
How to properly detect failing query?

### DISCUSSION THREAD

**Discussion by iBobik** [vuejs/vuefire#1378] [Questions and Help] [answered]
When some document in the collection is denied to read Firebase SDK fails:
> Uncaught (in promise) FirebaseError: Missing or insufficient permissions.
With this code:
```
getDocs(collection(useFirestore(), 'events'))
```

But when the same collection is read by this code:
```
useCollection<EventType>(collection(useFirestore(), 'events'))
```
It will not fail, just return empty list.

**Comment by posva** [comment]
You will need to use the Firebase Console and the Rules playground to check what permission is missing. Note this is not dependent on VueFire.

**Comment by iBobik** [comment]
I know what causes fail - I intentionally have multiple rules what allows to read only some documents.Firebase SDK throws exception when it happens, I just want Vuefire to pass it to my code so it can know it failed because of permissions. Currently it see just empty result.Honza Pobořil💬Signal2. 8. 2023 v 10:21, Eduardo San Martin Morote ***@***.***>:﻿
You will need to use the Firebase Console and the Rules playground to check what permission is missing. Note this is not dependent on VueFire.

**Reply by posva** [reply, marked as answer]
Ah, that's from the `error` variable: https://vuefire.vuejs.org/guide/realtime-data.html#Subscription-state

**Comment by iBobik** [comment]
Aha, that's it, thank you. Expected standard exception.Honza Pobořil💬Signal2. 8. 2023 v 14:39, Eduardo San Martin Morote ***@***.***>:﻿
Ah, that's from the error variable: https://vuefire.vuejs.org/guide/realtime-data.html#Subscription-state
